src/mattermostdriver/websocket.py

```diff
@@ -120,7 +120,7 @@ async def _authenticate_websocket(self, websocket, event_handler):
         """
         log.debug("Authenticating websocket")
         json_data = json.dumps({"seq": 1, "action": "authentication_challenge", "data": {"token": self._token}})
-        await websocket.send_bytes(json_data)
+        await websocket.send_str(json_data)
```

Comment:
Since you've modified to not change the JSON into bytes now, this should work. I wasn't sure what the `websockets` library did here. I bet even tho you encoded the JSON to bytes, send implicitly converted it to a string. Since we're now not changing to bytes, I would expect send_str to work.

https://api.mattermost.com/#tag/WebSocket makes no mention of encoding the JSON to bytes - So I think this is the correct change, but I was erring on changing as little as possible.

It's a bit for me to setup my test environment right now (due to $bigCompany reasons), and I'm not sure what version we are exactly (I also don't have admin access to check the version unless there is some API I couldn't quickly find to hit) and sending the encoded bytes definitely worked for me.

Reply:
@fried - Agree?

Reply:
yeah send_str would be correct here